empiric-package/empiric/publisher/async_client.py

```diff
@@ -0,0 +1,60 @@
+import asyncio
+from typing import List
+
+import aiohttp
+from empiric.core.entry import Entry
+from empiric.publisher.base import PublisherInterfaceT
+
+
+class EmpiricAsyncPublisherClient:
+    fetchers: List[PublisherInterfaceT] = []
+
+    def add_fetcher(self, fetcher: PublisherInterfaceT):
+        self.fetchers.append(fetcher)
+
+    async def fetch(self) -> List[Entry]:
+        tasks = []
+        async with aiohttp.ClientSession() as session:
+            for fetcher in self.fetchers:
+                data = fetcher.fetch(session)
+                tasks.append(data)
+            result = await asyncio.gather(*tasks)
+            return [val for subl in result for val in subl]
```

Comment:
Why do we need a separate client - the standard in starknet.py afaict is to always have `call` and `call_sync` methods on the same client

Reply:
this isn't calling, this is just fetching the data.  We could modify the client this is mainly to show what the workflow should be.  Happy to discuss how to integrate this with the API.